activerecord/lib/active_record/core.rb

```diff
@@ -134,6 +134,16 @@ def self.error_on_ignored_order_or_limit=(value)
 
       mattr_accessor :maintain_test_schema, instance_accessor: false
 
+      ##
+      # :singleton-method:
+      # Extra flags passed to database CLI tool (mysqldump/pg_dump) when calling db:structure:dump
+      mattr_accessor :structure_dump_flags, instance_accessor: false
```

Comment:
Yeah, I wasn't sure about `AR::Base` either. Should it be configurable with:

```ruby
ActiveRecord::Tasks::DatabaseTasks.structure_dump_flags = ["--noop"]
ActiveRecord::Tasks::DatabaseTasks.structure_load_flags = ["--noop"]
```

?